src/http-gateways/path-gateway.md

```diff
@@ -205,6 +205,30 @@ This is a URL-friendly alternative to sending an [`Accept`](#accept-request-head
 - `format=json` → `Accept: application/json`
 - `format=cbor` → `Accept: application/cbor`
 
+## Query Parameters for CAR Requests
+
+The following query parameters are only available for requests made with either a `format=car` query parameter or an `Accept: application/vnd.ipld.car` request header. These parameters modify shape of the IPLD graph returned within the car file
+
+### `car-scope` (request query parameter)
+
+Optional, `car-scope=(block|file|all)` with default value 'all', describes the shape of the dag fetched the terminus of the specified path whose blocks are included in the returned CAR file after the blocks required to traverse path segments.
+
+`block` - Only the root block at the end of the path is returned After blocks required to verify the specified path segments.
+
+`file` - For queries that traverse UnixFS data, `file` roughly means return blocks needed to verify the end of the path as a filesystem entity. In other words, all the blocks needed 'cat' a UnixFS file at the end of the specified path, or to 'ls' a UnixFS directory at the end of the specified path. For all queries that do not reference non-UnixFS data, `file` is equivalent to `block`
```

Comment:
## Is there a better name for this `car-scope` than `file` ?

Now is the last call to propose something better (cc @aschmahmann)
(some things i've heard: `blob`, `adl`, `bytes`, `data`)

Reply:
3 fairly separate bits of feeback related to this, I hope you don't mind me doing it here:

1. Specifically on this piece, the language I've adopted for explaining this is "unixfs entity"; but "entity" is the more generic form which translates to: if unixfs, then the _thing_ that's here: file or directory, all blocks of sharded file or directory, _or_ if not unixfs then just this block. I see the language in this doc has "entity" too so I'd suggest leaning in to that.

2. I've been wondering what this looks like if/when we escape our unixfs prison and have other ADLs at play. What if WNFS shows up with an ADL we can run that returns >1 block and it makes sense to include at this point. I think "entity" makes sense for that but I'm wondering how else we could make this more future-proof. Embedding the implicit unixfs seems like one problem we have and that special-case only shows up for "entity" and pathing and we do have an opportunity in this specific spec to do something about that—e.g. requiring an additional parameter to opt-in to unixfs special-casing, which would also make room for other ADLs. `bafyfoobar/path/to/thing?adl=unixfs&car-scope=entity`.

3. I'm not a fan of `car-scope`, because I don't see why we have to lock it in for "car". We're likely to have additional block containers in the future that we may not call a "car" (there's been >1 proposals for a CAR successor that are different enough to not naturally be a CARv3). But also, what the parameter is really doing is specifying a traversal depth and style. Even `dag-scope` would be a better name. I'd combine one of `dag`, `graph` or `traversal`, with one of `scope`, `type` or `mode`. "car" is the one that doesn't fit and it's already specified in the format.

Reply:
> I hope you don't mind me doing it here

Not at all, and I also tend to agree with all of the points you've made.

> entity

Sure, not attached to `shallow`, `node`, etc. I just want something that isn't obviously misleading.

> I'm not a fan of car-scope ... I'd combine one of dag, graph or traversal, with one of scope, type or mode. "car" is the one that doesn't fit and it's already specified in the format.

I like your style 😄 and agree with your assessment that CAR feels weird here because what's being described is a graph-ish operation not a "how are the bytes stored/transmitted" operation. The only counterpoint I could see here is that this feels format specific because the way we're expressing the various entities when describing how the graph should be sent to us in its rendered format uses HTTP Headers rather than query parameters which makes this a little bit format related.

For example `/ipfs/bafyfoo/path/to/directory?dag-scope=all` is invalid rather than say implying `format=tar` unless an alternative format like `car`, `iso`, `zip`,... was specified.

> I've been wondering what this looks like if/when we escape our unixfs prison and have other ADLs at play. What if WNFS shows up with an ADL we can run that returns >1 block and it makes sense to include at this point.

I too would very much like to escape UnixFS prison. To that extent if there is a good word meaning approximately "single logical unit of data" without any recursive properties it'd be nice to use that so that people are familiar with it if/when support for non-UnixFS multiblock data structures emerges.

However (and somewhat off topic, although I'd be interested to engage with it further if you'd like):
- I don't think `?adl=` is going to be what saves us here since you might have multiple ADLs to apply. Instead I suspect we end up with something a little closer to https://github.com/ipfs/specs/pull/293 but in some situations implemented as closer to `ipfs://bafyfoo.... rest of ipld uri semantics?path=ipld` so that we can have things like escape characters, multiple transformations in a given path, etc.
- I suspect that given the opportunity to try things again (e.g. with IPLD paths) that something like `ipfs://bafy-unixfs-dir;adl=unixfs/subdir?path=ipld&format=car` would by default have `dag-scope=entity` rather than `dag-scope=all` since that matches what we do in the non-car case.

Reply:
(went with `entity`, as Rod sugested – https://github.com/ipfs/specs/pull/402#issuecomment-1544819745)